src/words.ts

```diff
@@ -0,0 +1,127 @@
+type StepVector = [string, number][];
```

Comment:
Not a fan of this data type and how its literals appear in code. I recommend working with `Record<string, number>` instead.

Reply:
Where can I find `Record` documented?

Reply:
Where can I find an official documentation of `Record`?

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type

It's just JS objects without fixing the keys like types do.